nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -71,42 +64,34 @@ class MotionModel
    */
   virtual void predict(models::State & state)
   {
-    // Previously completed via tensor views, but found to be 10x slower
-    // using namespace xt::placeholders;  // NOLINT
-    // xt::noalias(xt::view(state.vx, xt::all(), xt::range(1, _))) =
-    //   xt::noalias(xt::view(state.cvx, xt::all(), xt::range(0, -1)));
-    // xt::noalias(xt::view(state.wz, xt::all(), xt::range(1, _))) =
-    //   xt::noalias(xt::view(state.cwz, xt::all(), xt::range(0, -1)));
-    // if (isHolonomic()) {
-    //   xt::noalias(xt::view(state.vy, xt::all(), xt::range(1, _))) =
-    //     xt::noalias(xt::view(state.cvy, xt::all(), xt::range(0, -1)));
-    // }
-
     const bool is_holo = isHolonomic();
     float max_delta_vx = model_dt_ * control_constraints_.ax_max;
     float min_delta_vx = model_dt_ * control_constraints_.ax_min;
     float max_delta_vy = model_dt_ * control_constraints_.ay_max;
     float max_delta_wz = model_dt_ * control_constraints_.az_max;
-    for (unsigned int i = 0; i != state.vx.shape(0); i++) {
-      float vx_last = state.vx(i, 0);
-      float vy_last = state.vy(i, 0);
-      float wz_last = state.wz(i, 0);
-      for (unsigned int j = 1; j != state.vx.shape(1); j++) {
-        float & cvx_curr = state.cvx(i, j - 1);
-        cvx_curr = std::clamp(cvx_curr, vx_last + min_delta_vx, vx_last + max_delta_vx);
-        state.vx(i, j) = cvx_curr;
-        vx_last = cvx_curr;
-
-        float & cwz_curr = state.cwz(i, j - 1);
-        cwz_curr = std::clamp(cwz_curr, wz_last - max_delta_wz, wz_last + max_delta_wz);
-        state.wz(i, j) = cwz_curr;
-        wz_last = cwz_curr;
+
+    unsigned int n_rows = state.vx.rows();
+    unsigned int n_cols = state.vx.cols();
+
+    // Default layout in eigen is column-major, hence accessing elements in
+    // column-major fashion to utilize L1 cache as much as possible
+    for (unsigned int i = 1; i != n_cols; i++) {
+      for (unsigned int j = 0; j != n_rows; j++) {
+        float vx_last = state.vx(j, i - 1);
+        float & cvx_curr = state.cvx(j, i - 1);
+        cvx_curr = std::min(vx_last + max_delta_vx, std::max(cvx_curr, vx_last + min_delta_vx));
```

Comment:
min/max use the `clamp` util (and in all other locations we still do a min/max manually